src/core/qgsfeaturerequest.h

```diff
@@ -101,6 +101,7 @@ class CORE_EXPORT QgsFeatureRequest
       GeometryNoCheck = 0, //!< No invalid geometry checking
       GeometrySkipInvalid = 1, //!< Skip any features with invalid geometry. This requires a slow geometry validity check for every feature.
       GeometryAbortOnInvalid = 2, //!< Close iterator on encountering any features with invalid geometry. This requires a slow geometry validity check for every feature.
+      GeometryFixInvalid = 3, //!< Try to fix encountered invalid geometry, otherwise skip feature.
```

Comment:
```suggestion
      GeometryFixInvalid = 3, //!< Try to fix encountered invalid geometry, otherwise skip feature. Since QGIS 3.18.
```

Hmm... I'm wondering about this "otherwise skip feature" choice. It doesn't sit right with me, and I'd prefer "otherwise abort". Or two modes, GeometryFixInvalidSkipOnFailure / GeometryFixInvalidAbortOnFailure.

@m-kuhn what do you think?

Reply:
Same here.
I like this pull request because it's an easy way out for operations where there are no hidden gotchas. For the more tricky datasets and operations we better give responsibility to the user and let him handle things with care.

Reply:
Adding an abort shouldn't be too tricky. I'll look into it.

Reply:
Should be done.